runtime/dict.go

```diff
@@ -390,7 +390,8 @@ func (d *Dict) Len() int {
 
 // putItem associates value with key in d, returning the old associated value if
 // the key was added, or nil if it was not already present in d.
-func (d *Dict) putItem(f *Frame, key, value *Object) (*Object, *BaseException) {
+// fun is defining the operation on d, which can be add or delete the entry.
+func (d *Dict) putItem(f *Frame, key, value *Object, fun func(f *Frame, d *Dict, entry *dictEntry, hash *Int, key, value *Object, index int) *BaseException) (*Object, *BaseException) {
```

Comment:
Hmm this didn't work out quite the way I hoped: the callback is pretty inelegant because of all the parameters it requires. Sorry to run around in circles on this, but I think we should find another way.

What do you think about passing a new `overwrite bool` parameter to putItem() that, when false, will do the setdefault behavior, i.e. will only write the value if no value is already present for that key.

Since we already return the original value, we can return it from setdefault.